mmengine/registry/registry.py

```diff
@@ -387,6 +387,10 @@ def get(self, key: str) -> Optional[Type]:
         The method will first parse :attr:`key` and check whether it contains
         a scope name. The logic to search for :attr:`key`:
 
+        - ``key`` represents the whole object name with its module
```

Comment:
If `key`` represents the whole object name with its module
        information, for example, `mmengine.model.BaseModel`, ``get``
        will directly return the class object :class:`BaseModel`.

        Otherwise, it will first parse :attr:`key` and check whether it contains
        a scope name. The logic to search for :attr:`key`:
        - ``key`` does not contain a scope name, i.e., it is purely a module
          name like "ResNet": :meth:`get` will search for ``ResNet`` from the
          current registry to its parent or ancestors until finding it.
        - ``key`` contains a scope name and it is equal to the scope of the
          current registry (e.g., "mmcls"), e.g., "mmcls.ResNet": :meth:`get`
          will only search for ``ResNet`` in the current registry.
        - ``key`` contains a scope name and it is not equal to the scope of
          the current registry (e.g., "mmdet"), e.g., "mmcls.FCNet": If the
          scope exists in its children, :meth:`get` will get "FCNet" from
          them. If not, :meth:`get` will first get the root registry and root
          registry call its own :meth:`get` method.